projects/cdk/directives/zoom/zoom.service.ts

```diff
@@ -41,12 +41,12 @@ export class TuiZoomService extends Observable<TuiZoomEvent> {
                             ),
                             map(({event, delta}) => {
                                 const clientX =
-                                    (event.touches[0].clientX +
-                                        event.touches[1].clientX) /
+                                    ((event.touches.item(0)?.clientX ?? 0) +
+                                        (event.touches.item(0)?.clientX ?? 0)) /
```

Comment:
Please be more attentive, second one should be using 1, not 0

Reply:
fixed